### ISSUE TITLE
[3.6] 1-1 Relationship doesn't return value

### ISSUE THREAD

**Issue by neilodiaz** [Laravel-Backpack/CRUD#2227]
# Bug report
1-1 Relationship doesn't return value.

Models:
**User**
- id
- name
- email
_..etc._

```
protected $fillable = [
    'name', 'email', 'password', 'username', 'active'
];
```

---------------------

**StudentInfo**
- id
- user_id
- course
_..etc._

```
/* 1-1 Relationship to User */
public function user() {
    return $this->belongsTo('\App\User');
}
```

### What I did
I followed this documentation:
https://backpackforlaravel.com/docs/3.6/crud-columns#text

```
$this->crud->addFields([
[
                'name' => 'user.email',
                'label' => 'Email',
                'type' => 'text'
            ],
]);
```
But doesn't return anything

### What I expected to happen
- It should return the `email`

### What happened
- Doesn't return anything.

### What I've already tried to fix it
Tried:
```
$this->crud->addFields([
[
                'name' => 'user',
                'label' => 'Email',
                'type' => 'text'
            ],
]);
```
This returns the json-encoded user fields:
``` {"id":22,"name":null,"email":"lyl_admin@example.org","username":"20190001","email_verified_at":null,"active":1} ```

### Backpack, Laravel, PHP, DB version
"php": "^7.1.3",
"backpack/crud": "^3.6",
"laravel/framework": "5.8.*",
DB Version: 10.1.39-MariaDB

When I run ```php artisan backpack:version``` the output is:

`Command "backpack:version" is not defined.`

**Comment by ApexMuse** [comment]
@neilodiaz  For the List operations you add Columns, and for the Create and Update operations you'll add Fields.

For the Create/Update operations you'll do this:

 ```
$this->crud->add**Fields**([
[
                'name' => '**email**',
                'label' => 'Email',
                'type' => 'text'
            ],
]);
```

For the List operation you'll do this:

$this->crud->add**Columns**([
[
                'name' => '**email**',
                'label' => 'Email',
                'type' => 'text'
            ],
]);

In your case you don't need to specify a relationship at all since you're not using one.  The Advanced Use Case you provided in the docs is for situations in which your user is referring to another (different) user, such as a parent user or a child user.

Hope this helps.

**Comment by ApexMuse** [comment]
@neilodiaz  Try changing the name of your relationship to parent() and using parent.email, like this:

```
/* 1-1 Relationship to User */
public function parent() {
    return $this->belongsTo('\App\User');
}
```

Also - for the list view change the method to addColumns()

this->crud->addColumns([
[
                'name' => 'parent.email',
                'label' => 'Email',
                'type' => 'text'
            ],
]);

**Comment by pxpm** [comment]
Hello guys,

Thanks for your help @ApexMuse, i hope changing names would solve it because changing to `addColumns` i am sure it will take no effect other than `addColumn`.

```
/**
     * Add multiple columns at the end of the CRUD object's "columns" array.
     *
     * @param array $columns
     */
    public function addColumns($columns)
    {
        if (count($columns)) {
            foreach ($columns as $key => $column) {
                $this->addColumn($column);
            }
        }
    }

```

@ApexMuse you have it working with different names in your project ? It's some bug you found and worked around it?

Let me know if changing name worked for you @neilodiaz .

I will have to dig deeper on why if that is the case. 

Best,
Pedro

**Comment by neilodiaz** [comment]
Hi @ApexMuse & @pxpm ,

Thank you for your responses.

Tried renaming the `public function user()` to `public function parent()` but still no value returned.

**Comment by neilodiaz** [comment]
Ok, I did some trial-and-error and I think I found a workaround here. Maybe it will also help others.

Updated it to:
```
$this->crud->addFields([
...
[
    'name' => 'email',
    'label' => 'Email',
    'type' => 'text'
],
...
]);
```
Notice the `'name' => 'email',`

And on my StudentInfo model, I added an Accesor:
```
public function getEmailAttribute() {
    return $this->user->email;
}
```
This now returns the related `email` value on the input field.

And to save the data, on the `update()` method on my `StudentInfoCrudController`, I added this:
```
...
if($request->email) {
    $user = $this->crud->entry->user()->first();
    $user->email = $request->email;
    $user->save();
}
...
```
I'm not sure if this is _correct_ way in doing it, but it gets the job done :)

Closing this now, but feel free to open it up again if you have any concerns.
